src/sas/qtgui/MainWindow/DataExplorer.py

```diff
@@ -983,11 +983,21 @@ def updatePerspectiveCombo(self, index):
         """
         Notify the gui manager about the new perspective chosen.
         """
+
+        # Notify via communicator
         self.communicator.perspectiveChangedSignal.emit(self.cbFitting.itemText(index))
-        self.chkBatch.setEnabled(self.parent.perspective().allowBatch())
-        # Deactivate and uncheck the swap data option if the current perspective does not allow it
-        self.chkSwap.setEnabled(self.parent.perspective().allowSwap())
-        if not self.parent.perspective().allowSwap():
+
+        # Set checkboxes
+        current_perspective = self.parent.perspective()
+
+        allow_batch = False if current_perspective is None else current_perspective.allowBatch()
+        allow_swap = False if current_perspective is None else current_perspective.allowSwap()
+
+        self.chkBatch.setEnabled(allow_batch)
+        self.chkSwap.setEnabled(allow_swap)
+
+        # Using this conditional prevents the checkbox for going into the "neither checked nor unchecked" state
+        if not allow_swap:
```

Comment:
Why would `allow_swap` not be either False or True? If you don't trust the return value of `current_perspective.allowSwap` then maybe there should be a test done on assigment in line 994?

Reply:
Allow swap is either True of False, yes. Not 100% sure what you're asking, the weirdness here is because the checkbox has three states, checked, unchecked, and square dot. Maybe using setChecked is the right way of avoiding this - currently, however, it's preserving the logic of the original code.